practice-one/src/components/Form/index.tsx

```diff
@@ -0,0 +1,199 @@
+// react
+import React, { useRef, useState } from "react";
+import { useForm, Controller } from 'react-hook-form';
+// css
+import "./Form.css";
+// components
+import Button from "../common/Button";
+import Input from "../common/Input";
+import DropdownSelect from "../Dropdown";
+// helpers
+import { uploadImage } from "@helpers/uploadImage";
+import { addStudentToAPI } from "@helpers/api";
+// constants
+import { GENDER_OPTIONS, CLASS_OPTIONS } from "@constants/dropdownData";
+
+export interface StudentFormProps {
+    closeForm: () => void;
+    onStudentAdd: () => void;
+}
+
+interface IFormInput {
+    name: string;
+    email: string;
+    phone: string;
+    gender: string;
+    password: string;
+    classes: string;
+    age: string;
+}
+
+const StudentForm: React.FC<StudentFormProps> = ({ closeForm, onStudentAdd }) => {
+    const {
+        control,
+        handleSubmit,
+        formState: { errors, isValid, isDirty }
+    } = useForm<IFormInput>({ mode: 'onChange' });
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const [imageFile, setImageFile] = useState<File | null>(null);
+    const [imagePreviewUrl, setImagePreviewUrl] = useState<string | null>(null);
+
+    const onSubmit = async (data: IFormInput) => {
+        let uploadedImageUrl = null;
+        if (imageFile) {
+            try {
+                const uploadResult = await uploadImage(imageFile);
+                uploadedImageUrl = uploadResult.data;
+            } catch (error) {
+                console.error("Error uploading image:", error);
+                return;
+            }
+        }
+
+        const newStudentData = {
+            ...data,
+            studentID: Math.floor(10000 + Math.random() * 90000),
+            imageUrl: uploadedImageUrl
+        };
+
+        try {
+            const result = await addStudentToAPI(newStudentData);
+            if (result) {
+                onStudentAdd();
+                closeForm();
+            } else {
+                console.error("Failed to add student");
```

Comment:
can you show the toast message or the text error to the user know create student is fail